xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/uicomponents/widgets/locationPicker.css

```diff
@@ -9,10 +9,18 @@
   margin-bottom: 0;
 }
 
+.location-picker .breadcrumb-container {
+  /*Allows overlapping with the location-action element with visual ordering in HTML.*/
+  float: left;
+  width: 100%;
+}
+
 .location-actions {
   float: right;
   /* Use the same padding as the breadcrumb. */
   padding: 8px 15px 0 15px;
+  /*Allows overlapping with the breadcrumb-container.*/
+  margin-left:-70px;
```

Comment:
I think this negative margin is a bad idea for two reasons:

1. The width of the actions is different for advanced vs. non-advanced users (one vs. two icons)
2. The negative margin means that the text of a very long space will be hidden behind the icons instead of continuing in the next line.

I would suggest using a Flexbox layout which should allow creating a layout where the first item fills the available space and the second item has just its natural width easily and without such hacks.

Reply:
I removed the negative margin.
Since there's overlapping of the two children components, I used some absolute positioning instead.

It doesn't seem to me that a Flexbox is useful in this case, is the patch okay without it?

Reply:
Just to illustrate what I mean, here is a screenshot after manually applying the absolute positioning:

![long and wrapping document reference hidden behind the buttons](https://user-images.githubusercontent.com/198317/212962710-4c595884-49d9-4e37-aa8a-6922d1e821e4.png)

Before any changes:

![long and wrapping document reference](https://user-images.githubusercontent.com/198317/212962980-093f3b0a-8ae6-4c9d-8702-fd6db1571747.png)
 
I don't insist on using Flexbox, it's just that Flexbox makes exactly this kind of layout very easy, with "this kind of layout" I mean two elements that share a line such that one of them has a fixed width while the other just takes the rest.